sph/include/sph/eos.hpp

```diff
@@ -18,33 +18,77 @@ namespace sph
  *
  * In this simple version of state equation, we calculate all depended quantities
  * also for halos, not just assigned particles in [startIndex:endIndex], so that
- * we can avoid halo exchange of p and c.
+ * we could potentially avoid halo exchange of p and c in return for exchanging halos of u.
```

Comment:
That's true probably only for the ideal gas EOS. But we are going to change it for a more complex one where there is no simple relation between these magnitudes.

Reply:
Yes I thought that this is the case and refrained from implementing this...